### ISSUE TITLE
The remote certificate is invalid according to the validation procedure.

### ISSUE THREAD

**Issue by Sivalingaamorthy** [IdentityServer/IdentityServer3#2864]
#Hi All,

I was able run IDS3 the IIS 8 with **id3srvtest.pfx**, which is default certificate found in the identityserver3, the problem I faced when I create my own certificate

I tried most of the solutions given in this form, but I think something is missed, can you suggest what's wrong with below steps .

Error I am getting, when I run the client.

[AuthenticationException: The remote certificate is invalid according to the validation procedure.]
[WebException: The underlying connection was closed: Could not establish trust relationship for the SSL/TLS secure channel.]

What is wrong with below steps.

1. Created Certificates (new.id3.host.com) suggested by Brock from the link https://brockallen.com/2015/06/01/makecert-and-creating-ssl-or-signing-certificates/ 
2. It created 2 certificates in the Certificate store (Local Computer)
      --  1 certificate was created as Server Authentication - Intended Purpose
      -- another 1 certificate was created as Code Signing - Intended Purpose
3. Exported the Server Authentication Certificate - it created .cer  (say new.id3.host.com.cer)   file.
4. Exported the Code Signing Certificate - it created another .pfx  (say csnewid3host.pfx) file.

     --  Added the certificate (new.id3.host.com.cer)  into Trusted Root Certificate Authority, Trusted People.

5. I put the csnewid3host.pfx in to the Host.Configuration.Config and made the file as Embedded Resource, and i changed the code accordingly.

6. Created the host website in iis 8, with the name and host name as new.id3.host.com 
7. Created the client website in iis 8, with the name and host name as new.id3.client.com. 

    -- Done with host file changes for the newly created sites.

8. Imported the new.id3.host.com.cer into IIS - Server Certificates
9. mapped the imported certificate to new.id3.host.com (website)
10. when I see the client website it got automatically the new.id3.host.com ssl certificate mapped [not sure why]

11. I compliled deployed the code for identityserver 3 host & client[was using sample MVC OWIN Client (Hybrid)].
12. when I browse https://new.id3.host.com/core, successful

13. when I browse https://new.id3.client.com, successful , I was able to see the login page.

14. After entering the user name & pwd (alice) and login button click, getting the above & subject mentioned error. What wrong on the above steps.

Your support is very much appreciated.

**Comment by brockallen** [comment]
> [AuthenticationException: The remote certificate is invalid according to the validation procedure.]
[WebException: The underlying connection was closed: Could not establish trust relationship for the SSL/TLS secure channel.]

This is a SSL error, not a signing certificate error and it's completely unrelated to IdentityServer. You need to get SSL working in your environment (IIS or IIS Express).

**Comment by Sivalingaamorthy** [comment]
Thanks Brock again. You are right, I missed few fundamentals of SSL.

I solved the issue.  The below may help to someone.

1. I messed up with Multiple certificates, so I cleaned up all.
2. I created 2 new certificates.

makecert -r -pe -n "CN=abchost.id3.com" -b 01/01/2000 -e 01/01/2036 -eku 1.3.6.1.5.5.7.3.1 -ss my -sr localMachine -sky exchange -sp "Microsoft RSA SChannel Cryptographic Provider" -sy 12 -len 2048

makecert -r -pe -n "CN=abcclient.id3.com" -b 01/01/2000 -e 01/01/2036 -eku 1.3.6.1.5.5.7.3.1 -ss my -sr localMachine -sky exchange -sp "Microsoft RSA SChannel Cryptographic Provider" -sy 12 -len 2048

3. Mapped the certificate to the websites, matching the certificate name & host name.

Caution:
4. I tried with wildcard certificate (like *.id3.com), for some reason it didn't work for me. Let me check again.
